__tests__/app.test.js

```diff
@@ -111,4 +112,212 @@ describe('GET requests', () => {
             })
         });
     });
+
+    describe('GET /api/articles/:article_id/comments', () => {
+        test('200 - responds with array of comments for the given article_id', () => {
+            return request(app)
+            .get('/api/articles/1/comments')
+            .expect(200)
+            .then(({body}) => {
+                expect(body.comments).toHaveLength(11)
+                body.comments.forEach((comment) => {
+                    expect(comment).toHaveProperty("comment_id", expect.any(Number))
+                    expect(comment).toHaveProperty("votes", expect.any(Number))
+                    expect(comment).toHaveProperty("created_at", expect.any(String))
+                    expect(comment).toHaveProperty("author", expect.any(String))
+                    expect(comment).toHaveProperty("body", expect.any(String))
+                    expect(comment).toHaveProperty("article_id", 1)
+                }) 
+                expect(body.comments).toBeSorted({ key: "created_at", descending: true })
+            })
+        })
+        test('200 - responds with message when given article_id valid but has no comments', () => {
+            return request(app)
+            .get('/api/articles/2/comments')
+            .expect(200)
+            .then(({body}) => {
+                expect(body.msg).toBe('article_id has no comments')
+            })
+        });
+        test('404 - article_id not found ', () => {
+            return request(app)
+            .get('/api/articles/999/comments')
+            .expect(404)
+            .then(({body}) => {
+                expect(body.msg).toBe('No articles found for article_id: 999')
+            })
+        })
+        test('400 - bad request', () => {
+            return request(app)
+            .get('/api/articles/itchy/comments')
+            .expect(400)
+            .then(({body}) => {
+                expect(body.msg).toBe('invalid request')
+            })
+        })
+    });
+
+
 });
+
+describe('POST', () => {
+    
+    describe('POST /api/articles/:article_id/comments', () => {
+        
+        test('201 - responds with the posted comment', () => {
+            const newComment = {body: 'Wonderful, simply wonderful', username: 'rogersop'}
+            return request(app)
+            .post('/api/articles/2/comments')
+            .send(newComment)
+            .expect(201)
+            .then(({ body }) => {
+                expect(body.comment).toMatchObject({
+                    comment_id: 19,
+                    body: 'Wonderful, simply wonderful',
+                    article_id: 2,
+                    author: 'rogersop',
+                    votes: 0
+                })
+                expect(body.comment).toHaveProperty("created_at")
+            })
+        });
```

Comment:
If you're using toMatchObject you can add a created_at key like this - 
`    expect(body.comment).toMatchObject({
                    comment_id: 19,
                    body: 'Wonderful, simply wonderful',
                    article_id: 2,
                    author: 'rogersop',
                    votes: 0, 
                    created_at: expect.any(String)
                })`

Reply:
Thank you!